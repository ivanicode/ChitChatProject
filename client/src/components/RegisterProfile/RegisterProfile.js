import React from 'react';

export function RegisterProfile(){

    //const {} = useRegisterProfileHooks()

    return (
        <form className="createProfileForm">
            <div className="createProfileDiv">
                <div className="formDiv">
                    <div className="greetings">
                        <h3>Witaj na portalu ChitChat</h3>
                        <h4>Wypełnij podstawowe dane, które wyświetlą się na Twoim profilu</h4>
                    </div>
                    <div className="nicknameInput">
                        <input type="text" placeholder="Wpisz swój nickname" />
                    </div>
                    <div className="cityInput">
                        <input type="text" placeholder="Nazwa Twojej miejscowości" />
                    </div>
                    <div className="genderRadioButton">
                        <p>Wybierz swoją płeć:</p>
                        <input type="radio" name="gender"/>Kobieta<br />
                        <input type="radio" name="gender"/>Mężczyzna<br />
                        <input type="radio" name="gender"/>Inna
                    </div>
                    <div className="addMainPicture">
                        <p>Wybierz swoje zdjęcie profilowe</p>
                        <input type="file" id="avatar" name="avatar" accept="image/png, image/jpeg"/>
                    </div>
                    <div className="interestsSelect">
                        <p>Wybierz min. jedno zainteresowanie (max 3)</p>
                        <select>
                            <option name="sport">Sport</option>
                            <option name="books">Książki</option>
                            <option name="series">Seriale</option>
                            <option name="travels">Podróże</option>
                            <option name="informatics">Informatyka</option>
                            <option name="diving">Nurkowanie</option>
                            <option name="cooking">Gotowanie</option>
                            <option name="music">Muzyka</option>
                            <option name="languages">Języki obce</option>
                            <option name="gaming">Gry online</option>
                            <option name="gardening">Uprawianie ogrodu</option>
                            <option name="fashion">Moda</option>
                            <option name="artistic">Plastyka</option>
                            <option name="automotive">Motoryzacja</option>
                            <option name="fishing">Wędkarstwo</option>
                            <option name="climbing">Wspinaczki</option>
                            <option name="horsmenship">Jeździectwo</option>
                            <option name="animals">Zwierzęta</option>
                            <option name="rekreationalSport">Sport rekreacyjny</option>
                            <option name="groupSport">Sporty grupowe</option>
                            <option name="extremeSports">Sporty ekstremalne</option>
                            <option name="combatSports">Sporty walki</option>
                            <option name="tattoos">Tatuaże</option>
                        </select>
                    </div>
                    <div className="relationshipButton">
                        <p>Jakiego rodzaju znajomości poszukujesz?</p>
                        <input type="radio" name="relationship"/>Relacji romantycznej<br />
                        <input type="radio" name="relationship"/>Relacji koleżeńskiej<br />
                        <input type="radio" name="relationship"/>Otwarta/y na obie relacje
                    </div>
                </div>
                <div className="buttonDiv">
                    <button type="submit" className="nextSubmit">Dalej</button>
                </div>
            </div>
    </form>
    )
}

export default RegisterProfile;
