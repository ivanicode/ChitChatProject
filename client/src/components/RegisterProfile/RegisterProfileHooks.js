import {useHistory} from 'react-router-dom';
import {useState} from 'react';
import { useSave } from '../../common/hooks/useSaveHook';

export function useProfileHooks(){

    const {
        formData,
        onChangeHandler
    } = useManageFormData()

    const {saveData} = useSave('/api/user/details')

    const {
        submitRegisterProfile
    } = useSubmitRegisterProfile(saveData, formData)

    return {
        submitRegisterProfile,
        formData,
        onChangeHandler
    }
}

export function useSubmitRegisterProfile(saveData, formData){
    const history = useHistory();
    function submitRegisterProfile(){
        saveData(formData)
        history.push('/pairing/chats');
    }
    return {submitRegisterProfile}
}


export const initialState = {
    nickname: '',
    city: '',
    gender: '',
    picture: '',
    interests: [],
    relationship: ''
}

export function useManageFormData(){

    const [formData, setFormData] = useState(initialState);

    function onChangeHandler(event){
        const options = event.target.options;
        const id = event.target.id;
        const value = event.target.value;
        const newState = {...formData};
        if(id === 'interests') {
            const checkedInterestsArray = Array.from(options).filter((element) => {
                return element.selected;
            }).map((el) => {
                return el.value;
            })
            if(checkedInterestsArray.length < 4){
                newState[id] = checkedInterestsArray;
            }
        } else {
            newState[id] = value;
        }
        setFormData(newState);
    }
    return {formData, onChangeHandler}
}


