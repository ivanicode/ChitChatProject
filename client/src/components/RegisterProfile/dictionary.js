export const gender = [
    {type: 'radio', name: 'gender', id: 1, label: 'Kobieta'},
    {type: 'radio', name: 'gender', id: 2, label: 'Mężczyzna'},
    {type: 'radio', name: 'gender', id: 3, label: 'Inna'},
]

export const hobby = [
    {id: 1, label: 'Sport'},
    {id: 2, label: 'Książki'},
    {id: 3, label: 'Seriale'},
    {id: 4, label: 'Podróże'},
    {id: 5, label: 'Informatyka'},
    {id: 6, label: 'Nurkowanie'},
    {id: 7, label: 'Gotowanie'},
    {id: 8, label: 'Muzyka'},
    {id: 9, label: 'Języki obce'},
    {id: 10, label: 'Gry online'},
    {id: 11, label: 'Uprawianie ogrodu'},
    {id: 12, label: 'Moda'},
    {id: 13, label: 'Plastyka'},
    {id: 14, label: 'Motoryzacja'},
    {id: 15, label: 'Wędkarstwo'},
    {id: 16, label: 'Wspinaczki'},
    {id: 17, label: 'Jeździectwo'},
    {id: 18, label: 'Zwierzęta'},
    {id: 19, label: 'Sport rekreacyjny'},
    {id: 20, label: 'Sporty grupowe'},
    {id: 21, label: 'Sporty ekstremalne'},
    {id: 22, label: 'Sporty walki'},
    {id: 23, label: 'Tatuaże'},
]               

export const relationship = [
    {type: 'radio', name: 'relationship', id: 1, label: 'Relacji romantycznej'},
    {type: 'radio', name: 'relationship', id: 2, label: 'Relacji koleżeńskiej'},
    {type: 'radio', name: 'relationship', id: 3, label: 'Otwarta/y na obie relacje'},
]