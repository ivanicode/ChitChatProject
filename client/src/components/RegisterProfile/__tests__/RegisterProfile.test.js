import React from 'react';
import {shallow} from 'enzyme';
import {RegisterProfile} from '../RegisterProfile';
import * as RegisterProfileHooks from '../RegisterProfileHooks';

describe('RegisterProfile component', () => {
    it('should match snapshot when no errors', () => {
        
        const wrapper = shallow(<RegisterProfile />)
        expect(wrapper).toMatchSnapshot()
    })
})