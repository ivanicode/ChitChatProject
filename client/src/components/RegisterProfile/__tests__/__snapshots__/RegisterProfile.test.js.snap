// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterProfile component should match snapshot when no errors 1`] = `
<form
  className="createProfileForm"
>
  <div
    className="createProfileDiv"
  >
    <div
      className="formDiv"
    >
      <div
        className="greetings"
      >
        <h3>
          Witaj na portalu ChitChat
        </h3>
        <h4>
          Wypełnij podstawowe dane, które wyświetlą się na Twoim profilu
        </h4>
      </div>
      <div
        className="nicknameInput"
      >
        <input
          placeholder="Wpisz swój nickname"
          type="text"
        />
      </div>
      <div
        className="cityInput"
      >
        <input
          placeholder="Nazwa Twojej miejscowości"
          type="text"
        />
      </div>
      <div
        className="genderRadioButton"
      >
        <p>
          Wybierz swoją płeć:
        </p>
        <input
          name="gender"
          type="radio"
        />
        Kobieta
        <br />
        <input
          name="gender"
          type="radio"
        />
        Mężczyzna
        <br />
        <input
          name="gender"
          type="radio"
        />
        Inna
      </div>
      <div
        className="addMainPicture"
      >
        <p>
          Wybierz swoje zdjęcie profilowe
        </p>
        <input
          accept="image/png, image/jpeg"
          id="avatar"
          name="avatar"
          type="file"
        />
      </div>
      <div
        className="interestsSelect"
      >
        <p>
          Wybierz min. jedno zainteresowanie (max 3)
        </p>
        <select>
          <option
            name="sport"
          >
            Sport
          </option>
          <option
            name="books"
          >
            Książki
          </option>
          <option
            name="series"
          >
            Seriale
          </option>
          <option
            name="travels"
          >
            Podróże
          </option>
          <option
            name="informatics"
          >
            Informatyka
          </option>
          <option
            name="diving"
          >
            Nurkowanie
          </option>
          <option
            name="cooking"
          >
            Gotowanie
          </option>
          <option
            name="music"
          >
            Muzyka
          </option>
          <option
            name="languages"
          >
            Języki obce
          </option>
          <option
            name="gaming"
          >
            Gry online
          </option>
          <option
            name="gardening"
          >
            Uprawianie ogrodu
          </option>
          <option
            name="fashion"
          >
            Moda
          </option>
          <option
            name="artistic"
          >
            Plastyka
          </option>
          <option
            name="automotive"
          >
            Motoryzacja
          </option>
          <option
            name="fishing"
          >
            Wędkarstwo
          </option>
          <option
            name="climbing"
          >
            Wspinaczki
          </option>
          <option
            name="horsmenship"
          >
            Jeździectwo
          </option>
          <option
            name="animals"
          >
            Zwierzęta
          </option>
          <option
            name="rekreationalSport"
          >
            Sport rekreacyjny
          </option>
          <option
            name="groupSport"
          >
            Sporty grupowe
          </option>
          <option
            name="extremeSports"
          >
            Sporty ekstremalne
          </option>
          <option
            name="combatSports"
          >
            Sporty walki
          </option>
          <option
            name="tattoos"
          >
            Tatuaże
          </option>
        </select>
      </div>
      <div
        className="relationshipButton"
      >
        <p>
          Jakiego rodzaju znajomości poszukujesz?
        </p>
        <input
          name="relationship"
          type="radio"
        />
        Relacji romantycznej
        <br />
        <input
          name="relationship"
          type="radio"
        />
        Relacji koleżeńskiej
        <br />
        <input
          name="relationship"
          type="radio"
        />
        Otwarta/y na obie relacje
      </div>
    </div>
    <div
      className="buttonDiv"
    >
      <button
        className="nextSubmit"
        type="submit"
      >
        Dalej
      </button>
    </div>
  </div>
</form>
`;
